backend/clap/src/main/java/zpi/ppea/clap/mappers/PartialRatingMapper.java

```diff
@@ -0,0 +1,44 @@
+package zpi.ppea.clap.mappers;
+
+import data_store.PartialRating;
+import data_store.PartialRatingRequest;
+import zpi.ppea.clap.dtos.PartialRatingDto;
+import zpi.ppea.clap.dtos.UpdatePartialRatingDto;
+
+public class PartialRatingMapper {
+    private PartialRatingMapper() {
+    }
+
+    public static PartialRatingDto partialRatingToDto(PartialRating partialRating) {
+        return PartialRatingDto.builder()
+                .partialRatingId(partialRating.getPartialRatingId())
+                .criterionId(partialRating.getCriterionId())
+                .points(partialRating.getPoints())
+                .justification(partialRating.getJustification())
+                .modified(partialRating.getModified())
+                .modifiedBy(partialRating.getModifiedBy())
+                .build();
+    }
+
+    public static PartialRating dtoToPartialRating(PartialRatingDto dto) {
+        return PartialRating.newBuilder()
```

Comment:
The amount of boilerplate code that can be generated is astounding. Remind me, what are the benefits of doing it this way instead of Mapstruct?

Reply:
Please consider these two version of methods in services:
<img width="529" alt="image" src="https://github.com/Veermove/zpi/assets/73716334/4abb334c-c810-4dea-9920-2bd881011f0a">
and 
<img width="550" alt="image" src="https://github.com/Veermove/zpi/assets/73716334/71f53333-7214-41d8-828f-ad8df962c700">
What is better? For me it's cleaner to put these 'mapping' to other class and not to service.

Reply:
You can define a method in the mapper interface for creating `RatingSubmissionResponseDto`? 
then the second method would look like
```java
var ratings = ratingsrepository.submissionRatings(submissionId, authenthication);
return DtoMapper.INSTANCE.RatingsResponseToDto(ratings)
```

You can go even further if you want to hide in another service.

Now how much cleaner is having these 4 files with code that can be easily generated:
![image](https://github.com/Veermove/zpi/assets/75857174/12b4ddd6-f486-488e-9901-1ad867741f90)
![image](https://github.com/Veermove/zpi/assets/75857174/87a1c913-e17d-4d14-b4e3-7b79345f3d47)

than having 1 file with same amount of functionality. 
![image](https://github.com/Veermove/zpi/assets/75857174/390fb730-4306-49f2-a9c6-f81c0095993e)





> If there is a change in entity I have to change them manually. I have to remember about it.

Yeah if only there existed a piece of software that could warn you that "Not all fields are mapped"